src/core/Akka.TestKit/TestKitBase_Receive.cs

```diff
@@ -670,15 +720,30 @@ public IReadOnlyCollection<object> ReceiveN(int numberOfMessages, TimeSpan max)
             return result;
         }
 
+        /// <inheritdoc cref="ReceiveN(int, TimeSpan)"/>
+        public async ValueTask<IReadOnlyCollection<object>> ReceiveNAsync(int numberOfMessages, TimeSpan max)
+        {
+            max.EnsureIsPositiveFinite("max");
+            var dilated = Dilated(max);
+            var result = await InternalReceiveNAsync(numberOfMessages, dilated, true).ToListAsync();
+            return result;
+        }
+
         private IEnumerable<object> InternalReceiveN(int numberOfMessages, TimeSpan max, bool shouldLog)
+        {
+            foreach(var msg in InternalReceiveNAsync(numberOfMessages, max, shouldLog).ToEnumerable())
+                yield return msg;
+        }
+
+        private async IAsyncEnumerable<object> InternalReceiveNAsync(int numberOfMessages, TimeSpan max, bool shouldLog)
```

Comment:
After reviewing your other changes, IAsyncEnumerable does not make any sense. If the returned IAsyncEnumerable is only going to be used internally, why use it at all? Change this to Task<IReadOnlyCollecton<T>> instead.

Reply:
To get that to work, I will have to hold `object` in memory. That won`t work with `yield`ing
![image](https://user-images.githubusercontent.com/8049896/155385907-a0c02c37-5bc3-447a-9671-784c5381904a.png)

Reply:
`IAsyncEnumerable` will work fine internally if we design the method such that we don't know how many messages we're receiveing and just want a forward-iterator. Checking the expected message count should be handled downstream in the methods that rest on top of `IAsyncEnumerable`

Reply:
So I think `IAsyncEnumerable` should work fine here - I'll take a look at some of this too

Reply:
hmmm... I thought it would be easier to return Task<IReadOnlyCollection<T>> instead of IAsyncEnumerable

Reply:
Your call